### ISSUE TITLE
python tests fail when run within a flux instance

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#309]
On current master 7a72f0095974d5751494158a2cfe566d64b2d4bf:
```
grondo@flux-core:~/workspace/src/bindings/python (master) $ ../../cmd/flux start --size=1 make check                                                                                             
flux-start: setrlimit: could not remove core file size limit: Operation not permitted
lt-flux-broker: [1438393593.090261] broker.info[0] nodeset: 0 (complete)
lt-flux-broker: [1438393593.090424] broker.info[0] starting shell
Making check in flux
make[1]: Entering directory `/home/ubuntu/workspace/src/bindings/python/flux'
Making check in core
make[2]: Entering directory `/home/ubuntu/workspace/src/bindings/python/flux/core'
make  check-local
make[3]: Entering directory `/home/ubuntu/workspace/src/bindings/python/flux/core'
echo Copying libraries to where they can be used by python for check
Copying libraries to where they can be used by python for check
[ "../../../../.." != "../../../../.." ] && cp  ../../../../../src/bindings/python/flux/core/*.py ./ || true
make[3]: Leaving directory `/home/ubuntu/workspace/src/bindings/python/flux/core'
make[2]: Leaving directory `/home/ubuntu/workspace/src/bindings/python/flux/core'
make[2]: Entering directory `/home/ubuntu/workspace/src/bindings/python/flux'
echo Copying libraries to where they can be used by python in-tree
Copying libraries to where they can be used by python in-tree
for LIB in _core.so _barrier.so _mrpc.so _kvs.so _jsc.so ; do \
            cp -f .libs/$LIB ./ ; \
        done
[ "../../../.." != "../../../.." ] && cp  ../../../../src/bindings/python/flux/*.py ./ || true
make  check-local
make[3]: Entering directory `/home/ubuntu/workspace/src/bindings/python/flux'
echo Copying libraries to where they can be used by python in-tree
Copying libraries to where they can be used by python in-tree
for LIB in _core.so _barrier.so _mrpc.so _kvs.so _jsc.so ; do \
            cp -f .libs/$LIB ./ ; \
        done
[ "../../../.." != "../../../.." ] && cp  ../../../../src/bindings/python/flux/*.py ./ || true
make[3]: Leaving directory `/home/ubuntu/workspace/src/bindings/python/flux'
make[2]: Leaving directory `/home/ubuntu/workspace/src/bindings/python/flux'
make[1]: Leaving directory `/home/ubuntu/workspace/src/bindings/python/flux'
make[1]: Entering directory `/home/ubuntu/workspace/src/bindings/python'
make  check-TESTS
make[2]: Entering directory `/home/ubuntu/workspace/src/bindings/python'
make[3]: Entering directory `/home/ubuntu/workspace/src/bindings/python'
FAIL: test_commands/test_runner.t 1 test.kvs.TestKVS.test_bad_init
FAIL: test_commands/test_runner.t 2 test.kvs.TestKVS.test_exists_dir
FAIL: test_commands/test_runner.t 3 test.kvs.TestKVS.test_exists_false
FAIL: test_commands/test_runner.t 4 test.kvs.TestKVS.test_exists_true
FAIL: test_commands/test_runner.t 5 test.kvs.TestKVS.test_fill
FAIL: test_commands/test_runner.t 6 test.kvs.TestKVS.test_iterator
FAIL: test_commands/test_runner.t 7 test.kvs.TestKVS.test_key_at
FAIL: test_commands/test_runner.t 8 test.kvs.TestKVS.test_kvs_dir_open
FAIL: test_commands/test_runner.t 9 test.kvs.TestKVS.test_read_non_existent
FAIL: test_commands/test_runner.t 10 test.kvs.TestKVS.test_read_non_existent_basedir
FAIL: test_commands/test_runner.t 11 test.kvs.TestKVS.test_remove
FAIL: test_commands/test_runner.t 12 test.kvs.TestKVS.test_set_deep
FAIL: test_commands/test_runner.t 13 test.kvs.TestKVS.test_set_dict
FAIL: test_commands/test_runner.t 14 test.kvs.TestKVS.test_set_float
FAIL: test_commands/test_runner.t 15 test.kvs.TestKVS.test_set_int
FAIL: test_commands/test_runner.t 16 test.kvs.TestKVS.test_set_list
FAIL: test_commands/test_runner.t 17 test.kvs.TestKVS.test_set_none
FAIL: test_commands/test_runner.t 18 test.kvs.TestKVS.test_set_string
FAIL: test_commands/test_runner.t 19 test.kvs.TestKVS.test_walk
FAIL: test_commands/test_runner.t 20 test.kvs.TestKVS.test_walk_with_no_handle
PASS: test_commands/test_runner.t 21 test.wrapper.TestWrapper.test_automatic_unwrapping
PASS: test_commands/test_runner.t 22 test.wrapper.TestWrapper.test_call_insufficient_arguments
PASS: test_commands/test_runner.t 23 test.wrapper.TestWrapper.test_call_invalid_argument_type
PASS: test_commands/test_runner.t 24 test.wrapper.TestWrapper.test_call_non_existant
PASS: test_commands/test_runner.t 25 test.wrapper.TestWrapper.test_masked_function
PASS: test_commands/test_runner.t 26 test.wrapper.TestWrapper.test_read_basic_value
PASS: test_commands/test_runner.t 27 test.wrapper.TestWrapper.test_set_pimpl_handle
ERROR: test_commands/test_runner.t - missing test plan
ERROR: test_commands/test_runner.t - exited with status 1
make[4]: Entering directory `/home/ubuntu/workspace/src/bindings/python'
Making all in flux
make[5]: Entering directory `/home/ubuntu/workspace/src/bindings/python/flux'
Making all in core
make[6]: Entering directory `/home/ubuntu/workspace/src/bindings/python/flux/core'
make[6]: Nothing to be done for `all'.
make[6]: Leaving directory `/home/ubuntu/workspace/src/bindings/python/flux/core'
make[6]: Entering directory `/home/ubuntu/workspace/src/bindings/python/flux'
echo Copying libraries to where they can be used by python in-tree
Copying libraries to where they can be used by python in-tree
for LIB in _core.so _barrier.so _mrpc.so _kvs.so _jsc.so ; do \
            cp -f .libs/$LIB ./ ; \
        done
[ "../../../.." != "../../../.." ] && cp  ../../../../src/bindings/python/flux/*.py ./ || true
make[6]: Leaving directory `/home/ubuntu/workspace/src/bindings/python/flux'
make[5]: Leaving directory `/home/ubuntu/workspace/src/bindings/python/flux'
make[5]: Entering directory `/home/ubuntu/workspace/src/bindings/python'
make[5]: Nothing to be done for `all-am'.
make[5]: Leaving directory `/home/ubuntu/workspace/src/bindings/python'
make[4]: Leaving directory `/home/ubuntu/workspace/src/bindings/python'
============================================================================
Testsuite summary for flux-core 0.1.0
============================================================================
# TOTAL: 29
# PASS:  7
# SKIP:  0
# XFAIL: 0
# FAIL:  20
# XPASS: 0
# ERROR: 2
============================================================================
See src/bindings/python/test-suite.log
============================================================================
make[3]: *** [test-suite.log] Error 1
make[3]: Leaving directory `/home/ubuntu/workspace/src/bindings/python'
make[2]: *** [check-TESTS] Error 2
make[2]: Leaving directory `/home/ubuntu/workspace/src/bindings/python'
make[1]: *** [check-am] Error 2
make[1]: Leaving directory `/home/ubuntu/workspace/src/bindings/python'
make: *** [check-recursive] Error 1
lt-flux-broker: [1438393594.232256] broker.info[0] 0: shutdown in 0.500s: Exited
flux-start: 0 (pid 30118) exited with rc=2
```

Full output:
```
=========================================================
   flux-core 0.1.0: src/bindings/python/test-suite.log
=========================================================

# TOTAL: 29
# PASS:  7
# SKIP:  0
# XFAIL: 0
# FAIL:  20
# XPASS: 0
# ERROR: 2

.. contents:: :depth: 2

ERROR: test_commands/test_runner
================================

TAP version 13
#Running doctests and unit tests in: /home/ubuntu/workspace/src/bindings/python/test/kvs.py
not ok 1 test.kvs.TestKVS.test_bad_init
FAIL: test_commands/test_runner.t 1 test.kvs.TestKVS.test_bad_init
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 2 test.kvs.TestKVS.test_exists_dir
FAIL: test_commands/test_runner.t 2 test.kvs.TestKVS.test_exists_dir
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 3 test.kvs.TestKVS.test_exists_false
FAIL: test_commands/test_runner.t 3 test.kvs.TestKVS.test_exists_false
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 4 test.kvs.TestKVS.test_exists_true
FAIL: test_commands/test_runner.t 4 test.kvs.TestKVS.test_exists_true
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 5 test.kvs.TestKVS.test_fill
FAIL: test_commands/test_runner.t 5 test.kvs.TestKVS.test_fill
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 6 test.kvs.TestKVS.test_iterator
FAIL: test_commands/test_runner.t 6 test.kvs.TestKVS.test_iterator
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 7 test.kvs.TestKVS.test_key_at
FAIL: test_commands/test_runner.t 7 test.kvs.TestKVS.test_key_at
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 8 test.kvs.TestKVS.test_kvs_dir_open
FAIL: test_commands/test_runner.t 8 test.kvs.TestKVS.test_kvs_dir_open
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 9 test.kvs.TestKVS.test_read_non_existent
FAIL: test_commands/test_runner.t 9 test.kvs.TestKVS.test_read_non_existent
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 10 test.kvs.TestKVS.test_read_non_existent_basedir
FAIL: test_commands/test_runner.t 10 test.kvs.TestKVS.test_read_non_existent_basedir
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 11 test.kvs.TestKVS.test_remove
FAIL: test_commands/test_runner.t 11 test.kvs.TestKVS.test_remove
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 12 test.kvs.TestKVS.test_set_deep
FAIL: test_commands/test_runner.t 12 test.kvs.TestKVS.test_set_deep
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 13 test.kvs.TestKVS.test_set_dict
FAIL: test_commands/test_runner.t 13 test.kvs.TestKVS.test_set_dict
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 14 test.kvs.TestKVS.test_set_float
FAIL: test_commands/test_runner.t 14 test.kvs.TestKVS.test_set_float
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 15 test.kvs.TestKVS.test_set_int
FAIL: test_commands/test_runner.t 15 test.kvs.TestKVS.test_set_int
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 16 test.kvs.TestKVS.test_set_list
FAIL: test_commands/test_runner.t 16 test.kvs.TestKVS.test_set_list
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 17 test.kvs.TestKVS.test_set_none
FAIL: test_commands/test_runner.t 17 test.kvs.TestKVS.test_set_none
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 18 test.kvs.TestKVS.test_set_string
FAIL: test_commands/test_runner.t 18 test.kvs.TestKVS.test_set_string
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 19 test.kvs.TestKVS.test_walk
FAIL: test_commands/test_runner.t 19 test.kvs.TestKVS.test_walk
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
not ok 20 test.kvs.TestKVS.test_walk_with_no_handle
FAIL: test_commands/test_runner.t 20 test.kvs.TestKVS.test_walk_with_no_handle
  ---
    message: |
      Traceback (most recent call last):
        File "/home/ubuntu/workspace/src/bindings/python/test/kvs.py", line 21, in setUp
          self.f = core.Flux()
        File "/home/ubuntu/workspace/src/bindings/python/flux/core/handle.py", line 36, in __init__
          handle = raw.flux_open(url, flags)
        File "/home/ubuntu/workspace/src/bindings/python/flux/wrapper.py", line 170, in __call__
          raise EnvironmentError(err, os.strerror(err))
      EnvironmentError: [Errno 2] No such file or directory
  ...
#Running doctests and unit tests in: /home/ubuntu/workspace/src/bindings/python/test/wrapper.py
ok 21 test.wrapper.TestWrapper.test_automatic_unwrapping
PASS: test_commands/test_runner.t 21 test.wrapper.TestWrapper.test_automatic_unwrapping
ok 22 test.wrapper.TestWrapper.test_call_insufficient_arguments
PASS: test_commands/test_runner.t 22 test.wrapper.TestWrapper.test_call_insufficient_arguments
ok 23 test.wrapper.TestWrapper.test_call_invalid_argument_type
PASS: test_commands/test_runner.t 23 test.wrapper.TestWrapper.test_call_invalid_argument_type
ok 24 test.wrapper.TestWrapper.test_call_non_existant
PASS: test_commands/test_runner.t 24 test.wrapper.TestWrapper.test_call_non_existant
ok 25 test.wrapper.TestWrapper.test_masked_function
PASS: test_commands/test_runner.t 25 test.wrapper.TestWrapper.test_masked_function
ok 26 test.wrapper.TestWrapper.test_read_basic_value
PASS: test_commands/test_runner.t 26 test.wrapper.TestWrapper.test_read_basic_value
Traceback (most recent call last):
  File "./test_commands/test_runner.t", line 53, in <module>
    ok 27 test.wrapper.TestWrapper.test_set_pimpl_handle
run_under_dir(result_set, '../test', 'test')
  File "./test_commands/test_runner.t", line 46, in run_under_dir
    run_tests_with_size(result_set, suite, size)
  File "./test_commands/test_runner.t", line 16, in run_tests_with_size
    with sideflux.run_beside_flux(size):
  File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
    PASS: test_commands/test_runner.t 27 test.wrapper.TestWrapper.test_set_pimpl_handle
return self.gen.next()
  File "/home/ubuntu/workspace/src/bindings/python/test_commands/sideflux.py", line 134, in run_beside_flux
    f.start()
  File "/home/ubuntu/workspace/src/bindings/python/test_commands/sideflux.py", line 96, in start
    raise EnvironmentError(self.sub.poll())
EnvironmentError: 1
close failed in file object destructor:
IOError: [Errno 9] Bad file descriptor
Exception OSError: (3, 'No such process') in <bound method SideFlux.__del__ of <sideflux.SideFlux object at 0x2b5c3b1b4290>> ignored
#Running doctests and unit tests in: /home/ubuntu/workspace/src/bindings/python/test/event.py
ERROR: test_commands/test_runner.t - missing test plan
ERROR: test_commands/test_runner.t - exited with status 1
```

**Comment by trws** [comment]
The python tests launch their own flux instance, so if that flux instance does not run they fail.  I'm actually not sure how best to handle this, should I have it clearing the flux environment from outside before attempting to launch the new instance?

**Comment by grondo** [comment]
Hm, the sharness tests work within a flux instance (now that flux-start always uses configuration file). So, in principle the new instance should run. 

 I would investigate further, but I was having trouble reaching root cause from python stack dump. If you have any hints I could follow up.

**Comment by trws** [comment]
Good point.  The python stack trace is not revealing anything useful 
because it's masking off the output of the underlying flux process by 
default until it has consumed the initial output to look for variables 
to scrape (which is not actually required anymore), and just failing 
when it can't open the socket to it on flux_open.

You can get more useful output by setting `SIDEFLUX_DEBUG` equal to 
something other than the empty string.

**Comment by grondo** [comment]
Ok, I tried to probe further but I'm not really getting anywhere. I did find I can run tests directly from the commandline (instead of `make check`), by using `flux env python test_commands/test_runner.t` which is handy.

However, setting `SIDEFLUX_DEBUG=t` in either the failing or succeeding case doesn't seem to produce any more output than without (unless I'm missing something). So it didn't help much, and I'm not familiar enough with python nor the various indirections in these tests to make much more progress.

This certainly isn't a critical issue, and I'll remember now to check for running in an instance if the python tests start mysteriously failing, so I'm willing to move on.

Maybe, however, for the future we need some kind of "sideflux works" sanity check before the main tests start running.
